File: Javascript/Arrays/D1_SecondLargest.js
function getSecondLargest(arr) {
      const n = arr.length;
      if (n < 2) return -1;
      
      let largest = -Infinity;
      let secondLargest = -Infinity;

      for (let i = 0 ; i < n ; i++) {
            if (arr[i] > largest) {
                  secondLargest = largest;
                  largest = arr[i];
            }
            else if (arr[i] > secondLargest && arr[i] !== largest) {
                  secondLargest = arr[i];
            }
      }

      return (secondLargest === -Infinity) ? -1 : secondLargest;
}

console.log(getSecondLargest([12, 35, 1, 10, 34, 1])); 
console.log(getSecondLargest([10]));
console.log(getSecondLargest([5, 5, 5])); 
console.log(getSecondLargest([-1, -5, -3])); 

File: CPP/Arrays/D1_SecondLargest.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int getSecondLargest(vector<int> &arr) {
      int n = arr.size();
      if (n < 2) return -1; 
  
      int largest = INT_MIN;
      int secondLargest = INT_MIN;
  
      for (int i = 0; i < n; i++) {
          if (arr[i] > largest) {
              secondLargest = largest;
              largest = arr[i];
          } else if (arr[i] > secondLargest && arr[i] != largest) {
              secondLargest = arr[i];
          }
      }
  
      return (secondLargest == INT_MIN) ? -1 : secondLargest;
}
  

int main() {
      vector<int> arr = {12, 35, 1, 10, 34, 1};
      int result = getSecondLargest(arr);
      cout << "Second largest element: " << result << endl;
      return 0;
}
